bin/node-template/node/src/service.rs

```diff
@@ -35,9 +35,9 @@ macro_rules! new_full_start {
 			.with_select_chain(|_config, backend| {
 				Ok(sc_client::LongestChain::new(backend.clone()))
 			})?
-			.with_transaction_pool(|config, client, _fetcher| {
+			.with_transaction_pool(|config, client, _fetcher, prometheus_registry| {
```

Comment:
I think this is the most straight-forward solution for this PR, but ugh this API. It doesn't make sense at all.
We should really remove these closures and simply have `prometheus_registry` be a local variable.

Reply:
I'm not requesting any change on this PR in particular, but I thought I'd comment on that in general.

Reply:
Have you seen this PR?
https://github.com/paritytech/substrate/pull/5557/files